MeetU/MeetU/Scripts/Controllers/Profile/profile-display.js

```diff
@@ -4,41 +4,147 @@
+                    }, function (privateProfile) {
+                        vm.privateProfile = privateProfile;
+                    }
+                ).$promise.then(function (data) {
+                    vm.infoHasLoaded = true;
+                    vm.profileInformation = getProfileInformation(data, vm.isPrivate);
+                });
+            } else {
+                vm.isPrivate = false;
+                PublicProfile.get({
+                    userId: $routeParams.profileId,
+                    joinedAmount: 3,
+                    launchedAmount: 3
+                }, function (publicProfile) {
+                    vm.publicProfile = publicProfile;
+                }
+                ).$promise.then(function (data) {
+                    vm.infoHasLoaded = true;
+                    vm.profileInformation = getProfileInformation(data, vm.isPrivate);
+                });
             }
-        ).$promise.then(function () {
-            $log.debug(vm.publicProfile);
-            vm.hasLoaded = true;
+        });
+
+        $q.all([
+            MeetupsLaunchedBy.query({ userId: $routeParams.profileId }, function (data) {
+                vm.launchedMeetups = data;
+            }).$promise,
+            MeetupsJoinedBy.query({ userId: $routeParams.profileId }, function (data) {
+                vm.joinedMeetups = data;
+            }).$promise
+        ]).then(function () {
+            vm.contentHasLoaded = true;
         })
 
-        
+        function showPersonalInformation() {
+            vm.isPersonalInformation = true;
+            vm.isLaunchedMeetups = false;
+            vm.isJoinedMeetups = false;
+            vm.activeMenu = "Personal Information";
+        }
+
+        function showLaunchedList() {
+            vm.isPersonalInformation = false;
+            vm.isLaunchedMeetups = true;
+            vm.isJoinedMeetups = false;
+            vm.activeMenu = "Launched Meetups";
+        }
+
+        function showJoinedList() {
+            vm.isPersonalInformation = false;
+            vm.isLaunchedMeetups = false;
+            vm.isJoinedMeetups = true;
+            vm.activeMenu = "Joined Meetups";
+        }
+
+        function init() {
+            vm.infoHasLoaded = false;
+            vm.contentHasLoaded = false;
+            vm.isPrivate = false;
+            vm.profileInformation = {};
+
+            vm.showPersonalInfo = showPersonalInformation;
+            vm.showLaunchedList = showLaunchedList;
+            vm.showJoinedList = showJoinedList;
+
+            vm.showJoinedList();
+        }
+
+        function getProfileInformation(profile, isPrivate) {
+            var genderIcon = (profile.gender === null || profile.gender === "male") ? "fa-mars" : "fa-venus",
+                genderColor = (profile.gender === null || profile.gender === "male") ? "color-male" : "color-female",
+                defaultPicture = "/Content/Images/dummyHead.PNG"
+
+            var profileInfo = {
```

Comment:
do you mean _.defaults(obj,{}) or var profileInfo = _.extends(profile, {})? I am pretty sure you can refactor this piece of code using one of them.